### PR TITLE
Fix Travis IPv6 and add k8s integration testing to CI

### PR DESCRIPTION
**Pull request by mrichmon**
Updating travis yaml file to:
- Force IPv6 to work in their VM environment
- Enable docker (requires VM environment and sudo)
- Run kubernetes integration tests in Travis

@miekg Please review, this impacts everyone using the CoreDNS codebase with Travis.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,39 @@
+sudo: required
+# Trusty distribution is much faster when sudo is required
+dist: trusty
```
Comment (miekg) [inline .travis.yml:3]: do we care?

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,39 @@
...
+# docker in the test environment. Docker is needed to launch a kubernetes
+# instance in the test environment.
+# (Dependencies are fun! :) )
+before_install:
+  - cat /proc/net/if_inet6
+  - uname -a
+  - sudo bash -c 'if [ `cat /proc/net/if_inet6 | wc -l` = "0" ]; then echo "Enabling IPv6" ; sysctl net.ipv6.conf.all.disable_ipv6=0 ; sysctl net.ipv6.conf.default.disable_ipv6=0 ; sysctl net.ipv6.conf.lo.disable_ipv6=0 ; fi'
+  - cat /proc/net/if_inet6
+  - env
+
```
Comment (miekg) [inline .travis.yml:26]: jeez.. but nice :) Is this some sort official way of doing it? Thinking of future breakage 'n stuff.

File: .travis.yml [hunk 3]
```diff
@@ -1,12 +1,39 @@
...
   - ./etcd-v2.3.1-linux-amd64/etcd &
   - go get
   - go get github.com/coreos/go-etcd/etcd
+  - if which docker &>/dev/null ; then docker pull gcr.io/google_containers/hyperkube-amd64:v1.2.4 ; docker ps -a ; fi
+  - pwd
+  - if which docker &>/dev/null ; then ./middleware/kubernetes/test/00_run_k8s.sh && ./middleware/kubernetes/test/10_setup_kubectl.sh && ./middleware/kubernetes/test/20_setup_k8s_services.sh ; docker ps -a ; fi
+
 script:
   - go test -tags etcd -race -bench=. ./...
+  - ./middleware/kubernetes/test/kubectl version && make testk8s
```
Comment (miekg) [inline .travis.yml]: why do we need these shell scripts? Can't we just start (and put everything) in the travis.yml? I get the k8s is more complex than just starting etcd, but littering the source with test scripts doesn't set well.

Also we can't test with `go test` in this setup, which is something I'm not willing to give up lightly.

File: .travis.yml (hunk 2 above)
Comment (mrichmon) [inline .travis.yml:26]: Nothing official.

I contacted Travis support and they claimed that IPv6 works in the VM environment but not in the container environment. I provided https://github.com/mrichmon/validate-travis-network to illustrate the issue. Two weeks later and no movement by Travis.
Also provided an update to their docs to correctly describe the behavior in https://github.com/travis-ci/docs-travis-ci-com/pull/662

With the conditional line in the script, if IPv6 is already enabled then nothing is altered. This should be fairly future-proof.

File: .travis.yml (hunk 3 above)
Comment (mrichmon) [inline .travis.yml]: The scripts allow easy creation of the same test system on the local machine. Can be moved into travis.yml with some pain. I wondered about putting the test scripts in a top-level test-support directory.

Moving to `go test` is easy. (New push is undergoing test.)

File: .travis.yml (hunk 1 above)
Comment (mrichmon) [inline .travis.yml:3]: sudo is required to use docker in the test environment.

Comment (miekg) [inline .travis.yml:3]: comment was actually about 'trusty', but this is minor

**Comment by miekg** [comment]
We should get rid of the shell scripts in kubernetes/test at some point. But LGTM.

**Pull request merged**